src/solver/application.cpp

```diff
@@ -421,14 +419,16 @@ void Application::readDataForTheStudy(Data::StudyLoadOptions& options)
     options.loadOnlyNeeded = true;
 
     // Load the study from a folder
-    Benchmarking::Timer loadTimer("Study loading", "study_loading", true, &pBenchmarkingContentHandler);
+    Benchmarking::Timer timer;
 
     if (study.loadFromFolder(pSettings.studyFolder, options) && !study.gotFatalError)
     {
         logs.info() << "The study is loaded.";
         logs.info() << LOG_UI_DISPLAY_MESSAGES_OFF;
     }
-    loadTimer.stop();
+
+    timer.stop();
+    pDurationCollector.addDuration("study_loading", timer.get_duration());
```

Comment:
[Style] You could pass `timer` as the 2nd argument, and call `get_duration` inside `addDuration`

Reply:
We could, but the code is clear as it is now. If we pass the `Timer` in the **addDuration** method and call stop() inside that method, it's shorter but less clear I think.

Reply:
We can keep the call to `Timer::stop` outside, but place the call to `Timer::get_duration` inside `addDuration`